### PR TITLE
feat(analyzer): enable host os info analyzer to support multiple nodes

### PR DESCRIPTION
**Pull request by DexterYan**
## Description, Motivation and Context
following PR https://github.com/replicatedhq/troubleshoot/pull/1608
- enable host info analyzer to analyzer multiple nodes
- save node list into `system/hostos_info_nodes.json`
- split collect results into `system/nodename/hostos_info.json`
- analyzer result and add node name in the title
- add tests

Demo Spec
```
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  name: sb
  runHostCollectorsInPod: true  # default is false 
spec:
  hostCollectors:
  - hostOS: {}
  hostAnalyzers:
  - hostOS:
        outcomes:
          - pass:
              when: "ubuntu >= 22.04"
              message: "Ubuntu 22.04 is supported"
          - fail:
              when: "ubuntu <= 16.04"
              message: "Ubuntu 16.04 is not supported"
```
Folders Structure:
Screenshot 2024-09-19 at 5 00 05 PM

Analyzer Result:
```
[
    {
        "name": "host.os.info.node.minikube",
        "labels": {
            "desiredPosition": "1",
            "iconKey": "",
            "iconUri": ""
        },
        "insight": {
            "name": "host.os.info.node.minikube",
            "labels": {
                "iconKey": "",
                "iconUri": ""
            },
            "primary": "Host OS Info - Node minikube",
            "detail": "Ubuntu 22.04 is supported",
            "severity": "debug"
        },
        "severity": "debug",
        "analyzerSpec": ""
    }
]
```

Multiple Nodes:
Screenshot 2024-09-20 at 10 08 59 AM

result:

```
[
    {
        "name": "host.os.info.node.multinode.demo.m02",
        "labels": {
            "desiredPosition": "1",
            "iconKey": "",
            "iconUri": ""
        },
        "insight": {
            "name": "host.os.info.node.multinode.demo.m02",
            "labels": {
                "iconKey": "",
                "iconUri": ""
            },
            "primary": "Host OS Info - Node multinode-demo-m02",
            "detail": "Ubuntu 22.04 is supported",
            "severity": "debug"
        },
        "severity": "debug",
        "analyzerSpec": ""
    },
    {
        "name": "host.os.info.node.multinode.demo.m02.node.multinode.demo",
        "labels": {
            "desiredPosition": "1",
            "iconKey": "",
            "iconUri": ""
        },
        "insight": {
            "name": "host.os.info.node.multinode.demo.m02.node.multinode.demo",
            "labels": {
                "iconKey": "",
                "iconUri": ""
            },
            "primary": "Host OS Info - Node multinode-demo-m02 - Node multinode-demo",
            "detail": "Ubuntu 22.04 is supported",
            "severity": "debug"
        },
        "severity": "debug",
        "analyzerSpec": ""
    }
]
```

## Checklist

- [x] New and existing tests pass locally with introduced changes.
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] The commit message(s) are informative and highlight any breaking changes
- [ ] Any documentation required has been added/updated. For changes to https://troubleshoot.sh/ create a PR [here](https://github.com/replicatedhq/troubleshoot.sh/pulls)

## Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

### CODE REVIEW

File: pkg/analyze/host_os_info.go [hunk 1]
```diff
@@ -27,30 +32,93 @@ func (a *AnalyzeHostOS) IsExcluded() (bool, error) {
...
 
+	// check if the host os info file exists (local mode)
 	contents, err := getCollectedFileContents(collect.HostOSInfoPath)
 	if err != nil {
-		return []*AnalyzeResult{&result}, errors.Wrap(err, "failed to get collected file")
+		//check if the host os info nodes file exists (remote mode)
+		contents, err := getCollectedFileContents(collect.HostOSNodes)
+		if err != nil {
+			return []*AnalyzeResult{&result}, errors.Wrap(err, "failed to get collected file")
+		} else {
```
Comment (nvanthao) [inline pkg/analyze/host_os_info.go]: nit: this else block can be removed.

**Review by nvanthao — CHANGES_REQUESTED** [review]
LGTM, only 1 nit suggest

File: pkg/analyze/host_os_info.go (hunk 1 above)
Comment (DexterYan) [reply pkg/analyze/host_os_info.go]: removed

**Comment by xavpaice** [comment]
the main thing to review here is that we have a ..._nodes.json that lists the nodes which were collected, and the analyzer then uses that to find the appropriate files.  We chose this over wildcard as it is more precise, a wildcard could pick up stuff we don't want if we later change the collector to add more files (like raw output).

**Comment by DexterYan** [comment]
Now the code has been improved to create only one file called `node_list.json` which has all the node name. It can be used for all host collectors without duplicated.

**Review by nvanthao — APPROVED** [review]
LGTM

**Pull request merged**